### ISSUE TITLE
Email settings and signature not saving 7.9.0

### ISSUE THREAD

**Issue by RobinHind** [salesagility/SuiteCRM#3639]
With the new email settings in the profile section, if you select a different editor or select a default email signature it doesn't save it. 

#### Issue
The email signature does not save on the user profile, once you compose an email the signature doesn't appear and when you go back to email settings it is declared as signatures: "none"

On my setup the email editor doesn't change either which may be connected?

#### Expected Behavior
When you select a signature on the email settings page and save it, it should add that signature to every new email

#### Actual Behavior
When you select an email signature and save it, it doesn't get added to a new email and the box reverts back to none

#### Steps to Reproduce
Open email settings under my profile, 
Change the signature box to a stored signature
click done, click save
open a new email - no signature

#### Context
For me this is a high priority but reading through the endless bugs effecting the new email module this is probably quite low priority.

On my install ive had to patch three separate issues already just to achieve some basic functionality like reading the email body, being able to reply to emails etc. this feels like a beta version?

#### Your Environment

* SuiteCRM Version used: 7.9.0
* Browser name and version (e.g. Chrome Version 58.0.3029.110 (64-bit) (64-bit)):
*  PHP7 MYSQL
* Operating System and version Ubuntu Linux 16.04.2

**Comment by Mausino** [comment]
@RobinHind i have same issue.. 
![draft](https://cloud.githubusercontent.com/assets/5773777/26730090/230869e4-47b0-11e7-92f8-9b2041bba4c1.png)

**Comment by dawansv** [comment]
I confirm this is a problem. This is what I was able to find out.

The settings are actually saved at first when just closing the Settings popup window. In fact if you do NOT click on Save on the main user profile page, then these settings are saved. You can navigate somewhere else, even log out and log in, go back to the settings and they are still saved. So it is the SAVE on the main profile page that seems to overwrite these settings.

These lines in modules/Users/Save.php seem to be the culprit:

```
        $GLOBALS['log']->debug("********** EMAIL 2.0 - Asynchronous - at: saveSettingsGeneral");
        $emailSettings = array();
        $emailSettings['emailCheckInterval'] = $_REQUEST['emailCheckInterval'];
        //$emailSettings['autoImport'] = isset($_REQUEST['autoImport']) ? '1' : '0';
        $emailSettings['alwaysSaveOutbound'] = '1';
        $emailSettings['sendPlainText'] = isset($_REQUEST['sendPlainText']) ? '1' : '0';
        $emailSettings['showNumInList'] = $_REQUEST['showNumInList'];
        $emailSettings['defaultOutboundCharset'] = $_REQUEST['default_charset'];
        $focus->setPreference('emailSettings', $emailSettings, '', 'Emails');

        // signature
        $focus->setPreference('signature_default', $_REQUEST['signature_id']);
        $focus->setPreference('signature_prepend', (isset($_REQUEST['signature_prepend'])) ? true : false);
```
These lines reset the email settings and signature from the $_REQUEST object but the $_REQUEST object from that page (the main profile page) only submit data from the main form, not the popup form whose data is submitted via some AJAX wizardry when it is closed. So these lines take empty values for all emails settings which then get saved back to empty/default.

I don't want to submit a patch to remove these lines as I don't know if they are needed from another form submission somewhere else, but if it is the case we do need to check if the values exists before setting them.

That said, fixing the email settings/signature save still does not solve the insert signature issue. As far as I can tell the signature should be inserted via javascript, but as the email interface as been redesigned I can't event see where that would happen at all. I need to find the time to first see how it is done in 7.8 so I even know where to look.

SuiteCRM is a great tool, and the efforts of the developers are much appreciated, but at this point I am wondering if with all the issues in the updated email module, it wouldn't be better to bring 7.9 back in beta and not advertise it on the main website. I am a new user of SuiteCRM and after playing with it for a few days I am going to use 7.8.4 for my production install. I'll keep testing 7.9 in parallel to help with bug fixes, but in my opinion I would not advise its use for production use yet.

**Comment by sravanitedla** [comment]
HI Team,

The issue is still der in 7.9.1. We are not able to get the signature on selecting the from address.

is it fixed or yet to be fixed.

Regards,
Sravani

**Comment by Mausino** [comment]
@sravanitedla we're also fighting with this issue... in version 7.9.1 is still apear. My developer will look on it and when i will have more info, i put them here :)

**Comment by Dillon-Brown** [comment]
@sravanitedla @Mausino Fresh installation of 7.9.1 shows this as fixed. Have you made sure to set the email signature within your personal email account settings?

![signature](https://user-images.githubusercontent.com/26431166/27324330-c962a59e-559c-11e7-9467-9e12feba504c.png)

**Comment by sravanitedla** [comment]
@Dillon-Brown 

Yes we have set signature with in personal account.
 We have tested this case by upgrading an older version of suitecrm to latest version i.e 7.9.1 as well we tested this case on fresh installation of 7.9.1.
 In the both cases we are able not able to get the signature appended while composing an email.


Regards,
Sravani.
